sys/src/9/amd64/archamd64.c

```diff
@@ -182,29 +209,26 @@ cpuidhz(uint32_t *info0, uint32_t *info1)
 		}
 		DBG("cpuidhz: 0x2a: %#llx hz %lld\n", rdmsr(0x2a), hz);
 	}
-	else if(memcmp(&info0[1], "AuthcAMDenti", 12) == 0){
+	else if(strcmp("AuthenticAMD",vendorid) == 0){
 		switch(info1[0] & 0x0fff0ff0){
 		default:
 			return 0;
 		case 0x00050ff0:		/* K8 Athlon Venice 64 / Qemu64 */
 		case 0x00020fc0:		/* K8 Athlon Lima 64 */
 		case 0x00000f50:		/* K8 Opteron 2xxx */
+		case 0x00100f60:		/* K8 Athlon II X2 */
 			msr = rdmsr(0xc0010042);
 			r = (msr>>16) & 0x3f;
 			hz = 200000000ULL*(4 * 2 + r)/2;
 			break;
-		case 0x00100f60:		/* K8 Athlon II */
-		case 0x00100f40:		/* Phenom II X2 */
+		case 0x00100f40:		/* Phenom II X2 && Athlon II X4 559 Processor */
```

Comment:
ah you don't need 100fc2, as the low nibl is cleared. 100fc0 will do it. But you need to find out what that is.